File: program52.c
//problem to get sum of even number of digits
#include<stdio.h>

int SumEvenDigits(int iNo)
{
    int iDigit=0,iSum=0;
    if(iNo<0)
    {
        iNo=-iNo;
    }
    while(iNo>0)        //this logic is same in all digit problems
    {
       iDigit=iNo%10;
       if(iDigit%2==0)
       iSum=iSum+iDigit;
       iNo=iNo/10;
    }  

    return iSum;

}

int main()
{
    int iValue=0,iRet=0;

    printf("Enter number\n");
    scanf("%d",&iValue);

    iRet=SumEvenDigits(iValue);
    printf("Summation of even digits are %d\n",iRet);

    return 0;
}

File: program52.java
//program to get sum of even digits in a number;
import java.lang.*;
import java.util.*;


class Count
{
    public int SumEvenDigit(int no)
    {
        int i=0,sum=0,digit=0;

        if(no<0)
        {
            no=-no;
        }

        while(no>0)
        {
            digit=no%10;
            if(digit%2==0)
            {
            sum=sum+digit;
            }
            
            no=no/10;
        }
        return sum;
       
    }
}


class program52
{
    public static void main(String arg[])
    {
        Count obj=new Count();
        int no=0,ret=0;
        Scanner sobj=new Scanner(System.in);
        System.out.println("Enter number");
        no=sobj.nextInt();
        ret=obj.SumEvenDigit(no);
        System.out.println("Sum of even digits is :"+ret);
      

       
        

    }
}